python/lsst/meas/algorithms/measureApCorr.py

```diff
@@ -164,103 +195,126 @@ def run(self, exposure, catalog):
+                                       (name, isGood.sum(), self.config.minDegreesOfFreedom + 1))
+
+            goodCat = goodRefCat[isGood].copy()
+
+            x = goodCat['slot_Centroid_x']
+            y = goodCat['slot_Centroid_y']
+            z = goodCat[self.refFluxNames.fluxName]/goodCat[fluxNames.fluxName]
+            ids = goodCat['id']
+
+            # We start with an initial fit that is the median offset; this
+            # works well in practice.
+            fitValues = np.median(z)
 
-            # If we don't have enough data points to constrain the fit, reduce the order until we do
             ctrl = self.config.fitConfig.makeControl()
-            while len(subset2) - ctrl.computeSize() < self.config.minDegreesOfFreedom:
-                if ctrl.orderX > 0:
-                    ctrl.orderX -= 1
-                if ctrl.orderY > 0:
-                    ctrl.orderY -= 1
-
-            # Fill numpy arrays with positions and the ratio of the reference flux to the to-correct flux
-            x = numpy.zeros(len(subset2), dtype=float)
-            y = numpy.zeros(len(subset2), dtype=float)
-            apCorrData = numpy.zeros(len(subset2), dtype=float)
-            indices = numpy.arange(len(subset2), dtype=int)
-            for n, record in enumerate(subset2):
-                x[n] = record.getX()
-                y[n] = record.getY()
-                apCorrData[n] = record.get(self.refFluxKeys.flux)/record.get(keys.flux)
-
-            for _i in range(self.config.numIter):
-
-                # Do the fit, save it in the output map
-                apCorrField = ChebyshevBoundedField.fit(bbox, x, y, apCorrData, ctrl)
-
-                if display:
-                    plotApCorr(bbox, x, y, apCorrData, apCorrField, "%s, iteration %d" % (name, _i), doPause)
-
-                # Compute errors empirically, using the RMS difference between the true reference flux and the
-                # corrected to-be-corrected flux.
-                apCorrDiffs = apCorrField.evaluate(x, y)
-                apCorrDiffs -= apCorrData
-                apCorrErr = numpy.mean(apCorrDiffs**2)**0.5
-
-                # Clip bad data points
-                apCorrDiffLim = self.config.numSigmaClip * apCorrErr
-                with numpy.errstate(invalid="ignore"):  # suppress NAN warning
-                    keep = numpy.fabs(apCorrDiffs) <= apCorrDiffLim
-                x = x[keep]
-                y = y[keep]
-                apCorrData = apCorrData[keep]
-                indices = indices[keep]
 
-            # Final fit after clipping
-            apCorrField = ChebyshevBoundedField.fit(bbox, x, y, apCorrData, ctrl)
+            allBad = False
+            for iteration in range(self.config.numIter):
+                resid = z - fitValues
+                # We allow floating-point slop to allow for field estimation
+                # noise in a completely flat field.
```

Comment:
Does this avoid a divide-by-zero somewhere?  I think that means a slightly longer comment here would be useful.

Reply:
I don't think so, but I've rephrased the comment to be clear this shows up in tests when the residual field is completely flat and median_abs_deviation doesn't return identically zero in that case.